src/features/broker-protection/actions/extract.js

```diff
@@ -128,12 +134,15 @@ export function createProfile (elementFactory, extractData) {
             output[key] = null
         } else {
             const elements = elementFactory(key, value)
-            const evaluatedValue = value?.findElements
-                ? findFromElements(elements, key, value)
-                : findFromElement(elements[0], key, value)
 
-            // Note: This can return a string, string[], or null
-            const extractedValue = extractValue(key, value, evaluatedValue)
+            // extract all strings first
+            const evaluatedValues = stringValuesFromElements(elements, key, value)
+
+            // clean them up - trimming, removing empties
+            const noneEmptyArray = cleanArray(evaluatedValues)
+
+            // Note: This can return any valid JSON valid, it depends on the extractor used.
+            const extractedValue = extractValue(key, value, noneEmptyArray)
```

Comment:
This is a bigger change than it looks. Previously the extract step would sometimes get an array, and sometimes a string. This made it difficult to reason about. Now all extractors receive a `string[]` for consistency